### DISCUSSION TITLE
Reading a reactive variable

### DISCUSSION THREAD

**Discussion by nnmm** [Textualize/textual#1249] [Q&A] [answered]
This assertion fails (in my app's `__init__()` function):

        self.ready = reactive(False)
        assert not self.ready

Am I using reactive variables wrong? The goal is to disable/enable a button based on this variable.

**Comment by davep** [comment, marked as answer]
By the looks of what you say here, you're creating the reactive inside your `__init__`. What you should do is declare it [at the class scope](https://textual.textualize.io/guide/reactivity/#reactive-attributes):

> To create these attributes import [reactive](https://textual.textualize.io/api/reactive/#textual.reactive.reactive) from textual.reactive, and assign them in the class scope.
